code/ui/components/tabs/Tabs.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using System.Collections.Generic;
+
+using Sandbox.UI;
+
+namespace TTTReborn.UI
+{
+    public partial class Tabs : TTTPanel
+    {
+        public readonly List<Tab> TabList = new();
+
+        public readonly Panel Header;
+        public readonly PanelContent PanelContent;
+
+        public Tab SelectedTab { get; private set; }
+
+        public Tabs() : base()
+        {
+            StyleSheet.Load("/ui/components/tabs/Tabs.scss");
+
+            Header = Add.Panel("header");
+
+            PanelContent = new PanelContent(this);
+            PanelContent.AddClass("content");
+        }
+
+        public Tab AddTab(string title, Action<PanelContent> createContent, Action onSelectTab = null)
```

Comment:
Similar to the `string` stuff, `Action onSelectTab` should just be an empty action instead of null. Then we don't ever need to null check in the future.

Reply:
I'm against not using null here. We never know, whether a function is empty then, resetting it would be horrible as well. Additionally, how would you set the default empty action as a param (global empty function shouldn't be the solution). Imo, checking null is better than running a useless empty function.

Reply:
nvm, but could still technically do something like this
https://stackoverflow.com/a/17337403

Reply:
no reason doing so because it's optional to  define that action.